### PR TITLE
feat: add Dockerfile and Makefile target to build UBI based sidecar container image

### PR DESCRIPTION
**Pull request by kkujawa-sumo**
example command to build image: 
```bash
make build-ubi TAG=ghcr.io/kkujawa-sumo/tailing-sidecar-ubi:test VERSION=0.3.1 RELEASE_NUMBER=6
```

### CODE REVIEW

File: sidecar/Makefile [hunk 1]
```diff
@@ -14,6 +14,10 @@ all: build push
 build:
 	docker build --tag $(TAG) --file ${root_dir}sidecar/Dockerfile ${root_dir}sidecar/
 
+build:
```
Comment (pmalek-sumo) [inline sidecar/Makefile:17]: This duplicates the target above

Comment (kkujawa-sumo) [reply sidecar/Makefile]: Thanks! I didn't add this change to the commit, now it is corrected.

File: sidecar/Makefile [hunk 2]
```diff
@@ -14,6 +14,10 @@ all: build push
 build:
 	docker build --tag $(TAG) --file ${root_dir}sidecar/Dockerfile ${root_dir}sidecar/
 
+build-ubi:
+	docker build --tag $(TAG) --build-arg VERSION=${VERSION} --build-arg RELEASE_NUMBER=${RELEASE_NUMBER} \
+	--file ${root_dir}sidecar/Dockerfile.ubi ${root_dir}sidecar/
```
Comment (pmalek-sumo) [inline sidecar/Makefile:19]: Formatting nitpick for readability

```suggestion
	docker build \
		--tag $(TAG) \
		--build-arg VERSION=${VERSION} \
		--build-arg RELEASE_NUMBER=${RELEASE_NUMBER} \
		--file ${root_dir}sidecar/Dockerfile.ubi ${root_dir}sidecar/
```

File: sidecar/Dockerfile.ubi [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+RUN make all
+
+FROM public.ecr.aws/sumologic/fluent-bit:1.6.10-ubi
+
+ARG VERSION=${VERSION}
+ARG RELEASE_NUMBER=${RELEASE_NUMBER}
+
+LABEL name="Tailing Sidecar" \
+      maintainer="collection@sumologic.com" \
+      vendor="sumologic" \
```
Comment (pmalek-sumo) [inline sidecar/Dockerfile.ubi:14]:
```suggestion
      vendor="SumoLogic" \
```

?

Comment (kkujawa-sumo) [reply sidecar/Dockerfile.ubi:14]: I think that we should write company name as "Sumo Logic". What do you think?

Comment (pmalek-sumo) [reply sidecar/Dockerfile.ubi]: 🤔  might be. We'd have to change the labels in other images as well

**Review by pmalek-sumo — APPROVED** [review]

**Pull request merged**